compiler/ilgen/IlBuilder.cpp

```diff
@@ -1494,9 +1494,8 @@ IlBuilder::integerizeAddresses(TR::IlValue **leftPtr, TR::IlValue **rightPtr)
    {
    TR::IlValue *left = *leftPtr;
    TR::IlValue *right = *rightPtr;
-   if (left->getSymbol()->getDataType() == TR::Address)
+   if (left->getSymbol()->getDataType() == TR::Address && right->getSymbol()->getDataType() == TR::Address)
```

Comment:
ok, i separated the handling of each operand, so that if either operand is an Address it will convert it to Word. I think that's a better solution. I don't want to encourage anyone to use this function, however, without good reason. We aren't missing all that many evaluators, I don't think, and the long term prospect for integerizeAddresses should be gleeful deletion.